packages/zoe/src/contractSupport/durability.js

```diff
@@ -100,3 +101,40 @@ export const provideChildBaggage = (baggage, category) => {
 };
 harden(provideChildBaggage);
 /** @typedef {ReturnType<typeof provideChildBaggage>} ChildBaggageManager */
+
+/**
+ * For use in contract upgrades to provide values that come from other vats.
+ * All vats must be able to finish their upgrade without contacting other vats,
+ * so whatever values an instance needs from other vats must be saved in the first
+ * incarnation and read from baggage in each subsequent.
+ *
+ * This abstracts that condition so that the contract can convert a dictionary of promises
+ * into a dictionary of values during its first prepare (start) and each additional
+ * automatically reads from the baggage.
```

Comment:
nice docs... an example would make it even nicer.

Reply:
good idea, done